src/main_thread/init/media_source_content_initializer.ts

```diff
@@ -990,12 +1008,39 @@ export interface IInitializeArguments {
     /** Behavior when a new video and/or audio codec is encountered. */
     onCodecSwitch: "continue" | "reload";
   };
+  /**
+   * When set to an object, enable "Common Media Client Data", or "CMCD".
+   */
+  cmcd?: ICmcdOptions | undefined;
```

Comment:
isn't `cmcd?` and `undefined` redundant ?  Even if there is slighlty difference in how typescript handle both

Reply:
Yeah it's not with TypeScript's `exactOptionalPropertyTypes` option which we're using and I want to allow callers to provide one (the absence of property) or the other (the property set to `undefined`).

Reply:
OK